memory/src/main/scala/filodb.memory/data/OffheapLFSortedIDMap.scala

```diff
@@ -442,13 +611,18 @@ extends OffheapLFSortedIDMapReader(memFactory, holderKlass) {
    * After this is called, concurrent modifications and reads of the map in inst will fail gracefully.
    */
   final def free(inst: MapHolder): Unit = {
-    var curState = state(inst)
-    while (curState != MapState.empty) {
-      val mapPtr = inst.mapPtr
-      if (casState(mapPtr, curState, MapState.empty))
-        if (UnsafeUtils.unsafe.compareAndSwapLong(inst, mapPtrOffset, mapPtr, 0))
-          memFactory.freeMemory(mapPtr)
-      curState = state(inst)
+    acquireExclusive(inst)
+    try {
+      var curState = state(inst)
+      while (curState != MapState.empty) {
+        val mapPtr = inst.mapPtr
+        if (casState(mapPtr, curState, MapState.empty))
```

Comment:
With an exclusive lock held, it seems likely unnecessary to do a CAS or check for a zero state, which in theory should never be possible with proper use of locks.  Though I guess extra safety doesn't hurt.

Reply:
I just decided to leave all the cas stuff in place for now, to keep the code change churn down.